build-tools/src/main/kotlin/org/jetbrains/kotlin/testing/native/NativeTest.kt

```diff
@@ -13,24 +13,33 @@ import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.tasks.*
 import org.jetbrains.kotlin.ExecClang
 import org.jetbrains.kotlin.bitcode.CompileToBitcode
+import org.jetbrains.kotlin.bitcode.CompileToBitcodeExtension
 import org.jetbrains.kotlin.konan.target.*
 
 open class CompileNativeTest @Inject constructor(
         @InputFile val inputFile: File,
-        @Input val target: String
+        @Input val target: String,
+        sanitizer: SanitizerKind,
 ) : DefaultTask() {
     @OutputFile
     var outputFile = project.buildDir.resolve("bin/test/$target/${inputFile.nameWithoutExtension}.o")
 
     @Input
     val clangArgs = mutableListOf<String>()
 
+    @Input
+    private val sanitizerFlags = when (sanitizer) {
+        SanitizerKind.NONE -> listOf()
+        SanitizerKind.ADDRESS -> listOf("-fsanitize=address")
```

Comment:
We can replace such `when`s with enum properties to avoid code duplication.

Reply:
TBH I think every property would be uniquely used, so I don't see code duplication:
* postfixes for build task names
* flags for the clang frontend
* flags for the clang backend (which just turned out to be the same as frontend flags)
* n flags for n different linkers